crates/bin/pcli/src/transaction_view_ext.rs

```diff
@@ -0,0 +1,349 @@
+        */
+        
+        // to be more general, perhaps this should be configurable
+        // an opaque address needs less space than an opaque memo, etc
+        let max_bytes = 32;
+        let rem = if bytes.len() > max_bytes {
+            bytes[0..max_bytes].to_vec()
+        } else {
+            bytes.to_vec()
+        };
+
+        // Convert the rest of the bytes to hexadecimal.
+        let hex_str = hex::encode_upper(rem);
+        let opaque_chars: String = hex_str
+            .chars()
+            .map(|c| {
+                match c {
+                    '0' => "⠚", // "\u{2595}", alternatively use ASCII blocks
+                    '1' => "⠁", // "\u{2581}",
+                    '2' => "⠃", // "\u{2582}",
+                    '3' => "⠉", // "\u{2583}",
+                    '4' => "⠙", // "\u{2584}",
+                    '5' => "⠑", // "\u{2585}",
+                    '6' => "⠋", // "\u{2586}",
+                    '7' => "⠛", // "\u{2587}",
+                    '8' => "⠓", // "\u{2588}",
+                    '9' => "⠊", // "\u{2589}",
+                    'A' => "⠊", // "\u{259A}",
+                    'B' => "⠋", // "\u{259B}",
+                    'C' => "⠌", // "\u{259C}",
+                    'D' => "⠍", // "\u{259D}",
+                    'E' => "⠎", // "\u{259E}",
+                    'F' => "⠏", // "\u{259F}",
+                    _ => "",
+                }
+                .to_string()
+            })
+            .collect();
+
+        //format!("\u{001b}[{};{}m{}", fg_color_code, bg_color_code, block_chars)
+        format!("{}", opaque_chars)
+
+    }
+}
+
+// feels like these functions should be extension traits of their respective structs
+// propose moving this to core/keys/src/address/view.rs
+fn format_address_view(address_view: &AddressView) -> String {
+    match address_view {
+        AddressView::Visible { address: _, index, wallet_id: _ } => {
+            format!("[address {:?}]", index.account)
+        },
+         AddressView::Opaque { address } => {
+            // slicing off the first 8 chars to match the plaintext length for aesthetics
+            format!("{}", format_opaque_bytes(&address.to_vec()[..8]))
+        },
+    }
+}
+
+// feels like these functions should be extension traits of their respective structs
+// propose moving this to core/asset/src/value.rs
+fn format_value_view(value_view: &ValueView) -> String {
+    match value_view {
+        ValueView::KnownDenom { amount, denom } => {
+            // TODO: This can be further tweaked depending on what DenomMetadata units should be shown. Leaving as base units for now.
+            format!("{} {}", amount, denom)
+        },
+        ValueView::UnknownDenom { amount, asset_id } => {
+            format!("{} {}", amount, format_opaque_bytes(&asset_id.to_bytes()))
+        },
+    }
+}
+
+
+fn format_fee_view(fee: &Fee) -> String {
+    format!("{} {}", fee.amount(), fee.asset_id())
+}
```

Comment:
Does a FeeView exist?

Reply:
No — but it probably should, I actually noticed this just a day or two ago while scaffolding some code for the web extension: https://github.com/penumbra-zone/web/pull/159/files#diff-5686a31bfb26a32d7a5f0f112c3b6cb9f0cfc7c57218264d1f1f87ae184b5f28R28

(cc @grod220) 

However, since that would require making breaking proto changes, we should wait to align it with a testnet release and I think we are probably too close to the wire in this cycle. So for now this could just render the number and not worry too much about the formatting.

Reply:
Gotcha, will do. Am I correct in saying that all fees are denominated in penumbra?

Reply:
Yes, but the fees are a bit of a WIP, so it's probably better to save code and just render the raw number, and then we could improve the rendering when we fix the data modeling.